File: quickSort.py
# -*- coding:utf-8 -*-

def quickSort(array):
    if len(array) < 2:
        return array
    else:
        pivot = array[0]    # 选择基准值
        # 由所有小于等于基准值的元素组成的子数组
        less = [i for i in array[1:] if i <= pivot]
        # 由所有大于基准值的元素组成的子数组
        greater = [i for i in array[1:] if i > pivot]
        return quickSort(less) + [pivot] + quickSort(greater)

File: golang/quickSort.go
package golang

func quickSort(arr []int) []int {
	if len(arr) < 2 {
		return arr
	}
	pivot := arr[0]
	less := []int{}
	greater := []int{}
	for _, v := range arr[1:] {
		if v <= pivot {
			less = append(less, v)
		} else {
			greater = append(greater, v)
		}
	}
	less, greater = quickSort(less), quickSort(greater)
	myarr := append(append(less, pivot), greater...)
	return myarr
}

// func main() {
// 	my_list := []int{1, 3, 9, 8, 7, 1}
// 	fmt.Println(quickSort(my_list))
// }
